docs/examples/05_fpl_intro.py

```diff
@@ -0,0 +1,461 @@
+
+# %%
+# By default, `fastplotlib` will enumerate the available devices and highlight which device has been selected by
+# default when importing the module.
+
+import fastplotlib as fpl
+import numpy as np
+from itertools import product
+
+# %%
+# ## Simple Image
+
+# create a `Figure` instance
+# by default the figure will have 1 subplot
+fig = fpl.Figure(size=(600, 500))
+
+# get a grayscale image
+data = iio.imread("imageio:camera.png")
+
+# plot the image data
+image_graphic = fig[0, 0].add_image(data=data, name="sample-image")
+
+# show the plot
+fig.show(sidecar=True)
+
+# %%
+# **Use the handle on the bottom right corner of the canvas to resize it.
+# You can also pan and zoom using your mouse!**
+#
+# Changing graphic **properties**
+
+image_graphic.cmap = "viridis"
+
+# %%
+# ### Slicing data
+#
+# **Most properties, such as data, support slicing!**
+#
+# Our image data is of shape [n_rows, n_columns]
+
+print(image_graphic.data.value.shape)
+
+# %%
+#
+
+image_graphic.data[::15, :] = 1
+image_graphic.data[:, ::15] = 1
+
+# %%
+# **Fancy indexing**
+
+image_graphic.data[data > 175] = 255
+
+# %%
+# **Adjust vmin vmax**
+
+image_graphic.vmin = 50
+image_graphic.vmax = 150
+
+# %%
+# **Reset the vmin vmax**
+
+image_graphic.reset_vmin_vmax()
+
+# %%
+# ### Change graphic properties {.strip-code,.keep-text}
+#
+# Now that we have a better idea of how graphic properties can be dynamically changed, let's practice :D
+#
+# **Question:** Can you change the data of the image to be random data of the same shape? Hint: Use `np.random.rand()`
+# to create the new data.
+
+# create new random dat of the same shape as the original image
+new_data = np.random.rand(*image_graphic.data.value.shape)
+
+# update the data
+image_graphic.data = new_data
+
+# %%
+# **Question:** Can you change the colormap of the image? See [here](https://matplotlib.org/stable/gallery/color/colormap_reference.html) for a list of colormaps.
+
+image_graphic.cmap = "hsv"
+
+# %% {.keep-code}
+#
+
+# close the plot
+fig.close()
+
+# %%
+# ### Image updates {.keep-code,.keep-text}
+# This examples show how you can define animation functions that run on every render cycle.
+
+# create another `Figure` instance
+fig_v = fpl.Figure(size=(600, 500))
+
+fig.canvas.max_buffered_frames = 1
+
+# make some random data again
+data = np.random.rand(512, 512)
+
+# plot the data
+fig_v[0,0].add_image(data=data, name="random-image")
+
+# a function to update the image_graphic
+# a plot will pass its plot instance to the animation function as an arugment
+
+def update_data(plot_instance):
+    new_data = np.random.rand(512, 512)
+    plot_instance[0,0]["random-image"].data = new_data
+
+#add this as an animation function
+fig_v.add_animations(update_data)
+
+fig_v.show(sidecar=True)
+
+# %%
+# ### Adding another animation function {.strip-code,.keep-text}
+#
+# **Question:** Can you add an animation function that toggles the colormap of the image between "plasma" and "hsv"?
```

Comment:
Not a very practical anim example?

Reply:
Suggest something else, maybe on data?

Reply:
we already have one with the data above, so I was trying to think of something else to do